src/sources/kubernetes_logs/mod.rs

```diff
@@ -596,6 +600,11 @@ impl Source {
             }
             None => ClientConfig::infer().await?,
         };
+        if let Ok(user_agent) = HeaderValue::from_str(&format!("{}/{}", PKG_NAME, PKG_VERSION)) {
+            client_config
+                .headers
+                .push((HeaderName::from_static("user-agent"), user_agent));
+        }
```

Comment:
Nit it seems like this is the real change and the rest is to handle the dependency updates. It might have been easier to review if you'd updated the dependencies in a separate PR. However, definitely appreciate you bumping them!

Reply:
The problem is that the `headers` field wasn't added to kube's Config struct until version 0.93.0: https://docs.rs/kube/0.93.0/kube/struct.Config.html#structfield.headers so bumping them was a prerequisite for the change I actually wanted to make here. Arguably they could still have been done first, but the motivation for it is easier to see when the changes are combined imo.

Reply:
👍 I figured it was something like that. I think separate PRs could have made reviews easier, but no big deal.